hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/storage/ContainerProtocolCalls.java

```diff
@@ -97,6 +98,28 @@ public final class ContainerProtocolCalls  {
   private ContainerProtocolCalls() {
   }
 
+  /**
+   * Creates a ContainerCommandRequestProto with version set.
+   */
+  public static ContainerCommandRequestProto.Builder getContainerCommandRequestProtoBuilder(int version) {
+    return getContainerCommandRequestProtoBuilder(null, version);
+  }
+
+  public static ContainerCommandRequestProto.Builder getContainerCommandRequestProtoBuilder() {
+    return getContainerCommandRequestProtoBuilder(null, ClientVersion.CURRENT.toProtoValue());
+  }
+
+  public static ContainerCommandRequestProto.Builder getContainerCommandRequestProtoBuilder(
+      ContainerCommandRequestProto req, int version) {
+    return (req == null ?
+        ContainerCommandRequestProto.newBuilder() : ContainerCommandRequestProto.newBuilder(req)).setVersion(version);
+  }
```

Comment:
`req` may already have `version` set.  `newBuilder(req)` preserves that in the builder, but `setVersion` overwrites it.  This is a problem if called via `getContainerCommandRequestProtoBuilder(req)`, without explicit version.

Reply:
If that is the case we should just send a builder than sending the request object. That is also one viable change.

Reply:
or we can just add a validation check and throw an exception if the version is not set.